<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TabWidget</class>
 <widget class="QTabWidget" name="TabWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>677</width>
    <height>485</height>
   </rect>
  </property>
  <property name="currentIndex">
   <number>1</number>
  </property>
  <widget class="QWidget" name="tabHardware">
   <attribute name="title">
    <string>Hardware</string>
   </attribute>
   <layout class="QFormLayout" name="formLayout">
    <property name="verticalSpacing">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <widget class="QLabel" name="label_4">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="text">
       <string>Keyboard &amp;model:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
      <property name="buddy">
       <cstring>keyboardModelComboBox</cstring>
      </property>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QComboBox" name="keyboardModelComboBox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>2</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="whatsThis">
       <string>Here you can choose a keyboard model. This setting is independent of your keyboard layout and refers to the &quot;hardware&quot; model, i.e. the way your keyboard is manufactured. Modern keyboards that come with your computer usually have two extra keys and are referred to as &quot;104-key&quot; models, which is probably what you want if you do not know what kind of keyboard you have.
</string>
      </property>
      <property name="maxVisibleItems">
       <number>15</number>
      </property>
      <property name="sizeAdjustPolicy">
       <enum>QComboBox::AdjustToContentsOnFirstShow</enum>
      </property>
     </widget>
    </item>
    <item row="1" column="0" colspan="2">
     <widget class="QWidget" name="lowerHardwareWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>100</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="margin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="tabLayouts">
   <attribute name="title">
    <string>Layouts</string>
   </attribute>
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_3">
      <item>
       <widget class="QGroupBox" name="switchingPolicyGroupBox">
        <property name="whatsThis">
         <string>If you select &quot;Application&quot; or &quot;Window&quot; switching policy, changing the keyboard layout will only affect the current application or window.</string>
        </property>
        <property name="title">
         <string>Switching Policy</string>
        </property>
        <layout class="QGridLayout">
         <item row="0" column="0">
          <widget class="QRadioButton" name="switchByGlobalRadioBtn">
           <property name="text">
            <string>&amp;Global</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">switchingPolicyButtonGroup</string>
           </attribute>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QRadioButton" name="switchByDesktopRadioBtn">
           <property name="text">
            <string>&amp;Desktop</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">switchingPolicyButtonGroup</string>
           </attribute>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QRadioButton" name="switchByApplicationRadioBtn">
           <property name="text">
            <string>&amp;Application</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">switchingPolicyButtonGroup</string>
           </attribute>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QRadioButton" name="switchByWindowRadioBtn">
           <property name="text">
            <string>&amp;Window</string>
           </property>
           <attribute name="buttonGroup">
            <string notr="true">switchingPolicyButtonGroup</string>
           </attribute>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="shortcutsGroupBox">
        <property name="title">
         <string>Shortcuts for Switching Layout</string>
        </property>
        <layout class="QGridLayout" name="_2">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Main shortcuts:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
           <property name="buddy">
            <cstring>xkbGrpShortcutBtn</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="xkbGrpShortcutBtn">
           <property name="whatsThis">
            <string>This is a shortcut for switching layouts which is handled by X.org. It allows modifier-only shortcuts.</string>
           </property>
           <property name="text">
            <string comment="no shortcut defined">None</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QToolButton" name="xkbGrpClearBtn">
           <property name="text">
            <string>…</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>3rd level shortcuts:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
           <property name="buddy">
            <cstring>xkb3rdLevelShortcutBtn</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="xkb3rdLevelShortcutBtn">
           <property name="whatsThis">
            <string>This is a shortcut for switching to a third level of the active layout (if it has one) which is handled by X.org. It allows modifier-only shortcuts.</string>
           </property>
           <property name="text">
            <string comment="no shortcut defined">None</string>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QToolButton" name="xkb3rdLevelClearBtn">
           <property name="text">
            <string>…</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Alternative shortcut:</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
           </property>
           <property name="buddy">
            <cstring>kdeKeySequence</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1" colspan="2">
          <widget class="KKeySequenceWidget" name="kdeKeySequence">
           <property name="whatsThis">
            <string>This is a shortcut for switching layouts. It does not support modifier-only shortcuts and also may not work in some situations (e.g. if popup is active or from screensaver).</string>
           </property>
           <property name="multiKeyShortcutsAllowed">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QCheckBox" name="showOSD_Chk">
           <property name="layoutDirection">
            <enum>Qt::RightToLeft</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <widget class="QGroupBox" name="layoutsGroupBox">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="title">
       <string>Configure layouts</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0" colspan="2">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="addLayoutBtn">
           <property name="text">
            <string>Add</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="removeLayoutBtn">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Remove</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="moveUpBtn">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Move Up</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="moveDownBtn">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Move Down</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="previewButton">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Preview</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QTableView" name="layoutsTableView">
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="showGrid">
          <bool>false</bool>
         </property>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QCheckBox" name="layoutLoopingCheckBox">
         <property name="text">
          <string>Spare layouts</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>15</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="1">
        <widget class="QGroupBox" name="layoutLoopingGroupBox">
         <property name="title">
          <string/>
         </property>
         <property name="flat">
          <bool>true</bool>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Main layout count:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QSpinBox" name="layoutLoopCountSpinBox"/>
          </item>
          <item row="0" column="2">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>431</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QWidget" name="tabAdvanced">
   <attribute name="title">
    <string>Advanced</string>
   </attribute>
   <layout class="QGridLayout" name="gridLayout_1">
    <item row="0" column="0" colspan="2">
     <widget class="QCheckBox" name="configureKeyboardOptionsChk">
      <property name="text">
       <string>&amp;Configure keyboard options</string>
      </property>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QTreeView" name="xkbOptionsTreeView">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="headerHidden">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KKeySequenceWidget</class>
   <extends>QWidget</extends>
   <header>kkeysequencewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
 <slots>
  <signal>changed(bool)</signal>
 </slots>
 <buttongroups>
  <buttongroup name="switchingPolicyButtonGroup"/>
 </buttongroups>
</ui>
